lib/thrift/ast.ex

```diff
@@ -66,25 +76,27 @@ defmodule Thrift.AST do
     defstruct line: nil, name: nil, value: nil, type: nil
 
     @spec new(charlist, Literals.t(), Types.t()) :: t
-    def new(name, val, type) do
-      %Constant{name: atomify(name), value: cast(type, val), type: type}
+    def new(name, value, type) do
+      %Constant{name: atomify(name), value: cast(type, value), type: type}
     end
   end
 
   defmodule TEnum do
-    @moduledoc false
-    @type enum_value :: bitstring | integer
+    @moduledoc """
+    An enumerated type with named values.
+    """
+    @type value :: non_neg_integer()
```

Comment:
Is this true? Does thrift enforce enum values to be positive? So I couldn't do:
```
enum Weirdo {
  CATS = -1
  DOGS = 0
}
```

Reply:
That's what the [Thrift IDL spec](https://thrift.apache.org/docs/idl#enum) says:

> An enum creates an enumerated type, with named values. If no constant value is supplied, the value is either 0 for the first element, or one greater than the preceding value for any subsequent element. Any constant value that is supplied must be non-negative.

... although the Apache Thrift compiler just reads the value as an `int32_t` without a range check. I'm fine with changing this back to `integer()` because we don't have any restrictions on the value range ourself.